sdk/typescript/src/providers/provider.ts

```diff
@@ -193,98 +193,18 @@ export abstract class Provider {
   abstract syncAccountState(address: string): Promise<any>;
 
   /**
-   * Get events for one transaction
-   * @param digest transaction digest to search by
-   * @param count max result count
+   * Get events for a given query criteria
+   * @param query - the transaction query criteria.
+   * @param cursor - optional paging cursor
+   * @param limit - maximum number of items per page
+   * @param order - transaction query ordering
    */
-  abstract getEventsByTransaction(
```

Comment:
Since we cut the commit for DevNet 0.12.0, which does not include this commit, and we would like to do an SDK release later this week, can you mark these methods as `@deprecated` instead of deleting them? Otherwise the SDK will lose the capability to query any events since the new `sui_getEvents` endpoint will not be available until the release next week

Reply:
Will it be easier if we commit after the next cut?